jcansdale/ghvs
# Add support for opening a GitHub URL in VS Code

This adds support for navigating from pull request comments and files on dotcom, to the equivalent repository, file and line in Visual Studio. If the repository is already open in Visual Studio, the file will be opened and navigated in the live instance.

## What this PR does

- Navigate to the line an in-line comment is on in Visual Studio
- Navigate to a line from the pull request file view in Visual Studio

## How to test

1. Add a bookmarklet called `Open in VS` to your browser with the following script:
```
javascript:window.location.href='x-github-client://openRepo/'+window.location.href
```
2. Select a line in the pull request file view:
![image](https://user-images.githubusercontent.com/11719160/68216784-63de2600-ffd9-11e9-9204-12f2eb9500fe.png)
3. Click on the `Open in VS` bookmarklet

Expect the following:

1. If there's a running instance of Visual Studio on the repository it will be activated
2. The targeted file will be opened
3. The highlighted line or range will be selected
